controllers/api/v1/keyword_test.go

```diff
@@ -276,4 +276,114 @@ var _ = Describe("KeywordController", func() {
 			})
 		})
 	})
+
+	Describe("POST /api/v1/keyword/upload", func() {
+		Context("given a valid access token", func() {
+			Context("given a valid upload file", func() {
+				It("returns status 204", func() {
+					oauthClient := FabricateOauthClient(faker.UUIDHyphenated(), faker.Password())
+					user := FabricateUser(faker.Email(), faker.Password())
+					accessToken := FabricatorAccessToken(oauthClient.ID, user.Id)
+					headers, body := CreateMultipartRequestInfo("tests/fixtures/files/valid.csv", "text/csv")
+					headers.Add("Authorization", fmt.Sprintf("Bearer %v", accessToken))
+
+					response := MakeAuthenticatedRequest("POST", "/api/v1/keyword/upload", headers, body, nil)
+
+					Expect(response.StatusCode).To(Equal(http.StatusNoContent))
+				})
+
+				It("creates keyword list", func() {
+					oauthClient := FabricateOauthClient(faker.UUIDHyphenated(), faker.Password())
+					user := FabricateUser(faker.Email(), faker.Password())
+					accessToken := FabricatorAccessToken(oauthClient.ID, user.Id)
+					headers, body := CreateMultipartRequestInfo("tests/fixtures/files/valid.csv", "text/csv")
+					headers.Add("Authorization", fmt.Sprintf("Bearer %v", accessToken))
+					queryList := map[string]interface{}{"user_id": user.Id}
+
+					totalRowsBeforeRequest, err := models.CountAllKeyword(queryList)
+					if err != nil {
+						Fail(fmt.Sprintf("Count all keyword failed: %v", err.Error()))
+					}
+
+					_ = MakeAuthenticatedRequest("POST", "/api/v1/keyword/upload", headers, body, nil)
+
+					totalRowsAfterRequest, err := models.CountAllKeyword(queryList)
+					if err != nil {
+						Fail(fmt.Sprintf("Count all keyword failed: %v", err.Error()))
+					}
+
+					Expect(totalRowsAfterRequest - totalRowsBeforeRequest).To(BeNumerically(">", 0))
+				})
+			})
+
+			Context("given an INVALID upload file", func() {
+				It("returns status 500", func() {
```

Comment:
Should it be status `422 UnprocessableEntity` as this is the error caused by the user's input?